cg/apps/gisaid/gisaid.py

```diff
@@ -0,0 +1,126 @@
+"""Interactions with the gisaid cli upload"""
+
+import logging
+from pathlib import Path
+from typing import List
+
+from cg.store import models
+from cg.meta.workflow.fastq import FastqHandler
+from ...utils import Process
+from .models import GisaidSample, UpploadFiles
+from housekeeper.store import models as hk_models
+
+from cg.meta.meta import MetaAPI
+
+LOG = logging.getLogger(__name__)
+
+import csv
+
+
+class GisaidAPI(MetaAPI):
+    """Interface with Gisaid cli uppload"""
+
+    def __init__(self, config: dict):
+        super().__init__(config)
+        self.gisaid_submitter = config["gisaid"]["submitter"]
+        self.gisaid_binary = config["gisaid"]["binary_path"]
+        self.process = Process(binary=self.gisaid_binary)
+        self.fastq_handler = FastqHandler()
+
+    def get_headers(self) -> list:
+        """Get GisaidSample keys as list."""
+
+        return list(GisaidSample.schema().get("properties").keys())
+
+    def get_sample_row(self, sample: models.Sample, family_id: str, fasta_header: str) -> List[str]:
+        """Build row for a sample in the batch upload csv."""
+
+        orig_lab: str = "Stockholm"  # sample.originating_lab
+        collection_date: str = "201122"  # sample.collection_date str or date, we dont know yet
+        gisaid_sample = GisaidSample(
+            covv_virus_name=fasta_header,
+            covv_subm_sample_id=sample.name,
+            submitter=self.gisaid_submitter,
+            fn=f"{family_id}.fasta",
+            covv_collection_date=collection_date,
+            lab=orig_lab,
+        )
+        return [gisaid_sample.dict().get(header) for header in self.get_headers()]
```

Comment:
if get headers are same sorting order as gisaid_sample.dict, does it mean you only want a list of dict values?

Reply:
Yes. I want a list of dict values in a specific order

Reply:
Agreed on going back to use HEADER constant